packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -1164,9 +1176,12 @@ class SelectionOverlay {
     if (!listEquals(_selectionEndpoints, value)) {
       markNeedsBuild();
       if (_isDraggingEndHandle || _isDraggingStartHandle) {
+
         switch(defaultTargetPlatform) {
           case TargetPlatform.android:
-            HapticFeedback.selectionClick();
+            if (hapticFeedbackEnabled) {
+              HapticFeedback.selectionClick();
```

Comment:
There is other place that call this method as well. we need to guard them behind the setting.

Reply:
I found selectable_region and drag_target as additional places that call HapticFeedback.selectionClick(). What is interesting is that neither of those location check the running os.

Reply:
both iOS and Android will vibrate when doing longpress select, and desktops have not implemented vibrate. That is why it work without checking OS. We should check it now since Android will behave different.